src/monitor.c

```diff
@@ -458,3 +470,59 @@ struct wlr_box monitor_get_active_geom(struct monitor *m)
     struct wlr_box geom = tag_get_active_geom(tag);
     return geom;
 }
+
+void handle_output_mgr_apply(struct wl_listener *listener, void *data)
+{
+	struct wlr_output_configuration_v1 *config = data;
+	handle_output_mgr_apply_test(config, false);
+}
+
+// apply_output_config
+void handle_output_mgr_apply_test(struct wlr_output_configuration_v1 *config, bool test)
+{
+	struct wlr_output_configuration_head_v1 *config_head;
+	bool ok = true;
+
+	wl_list_for_each(config_head, &config->heads, link) {
+		struct wlr_output *wlr_output = config_head->state.output;
+
+		wlr_output_enable(wlr_output, config_head->state.enabled);
+
+		if (config_head->state.enabled) {
+			if (config_head->state.mode)
+				wlr_output_set_mode(wlr_output, config_head->state.mode);
+			else
+				wlr_output_set_custom_mode(wlr_output,
+						config_head->state.custom_mode.width,
+						config_head->state.custom_mode.height,
+						config_head->state.custom_mode.refresh);
+
+			wlr_output_layout_move(server.output_layout, wlr_output,
+					config_head->state.x, config_head->state.y);
+			wlr_output_set_transform(wlr_output, config_head->state.transform);
+			wlr_output_set_scale(wlr_output, config_head->state.scale);
+		}
+
+		if (test) {
+			ok &= wlr_output_test(wlr_output);
+			wlr_output_rollback(wlr_output);
+		} else
```

Comment:
if you have used braces in the if/else construct once you must use it on the other ones too. That means use else {} not else without braces